src/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,36 @@
+import type { Chain } from 'viem';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import type { WalletCapabilities } from './types';
+
+export function useCapabilitiesSafe({
+  chain,
+}: {
+  chain: Chain;
+}): WalletCapabilities {
+  const { connector, isConnected } = useAccount();
+
+  // Metamask doesn't support wallet_getCapabilities
+  const isMetamaskWallet = connector?.id === 'io.metamask';
+  const enabled = isConnected && !isMetamaskWallet;
+
+  const { data: capabilities } = useCapabilities({ query: { enabled } });
+
+  const atomicBatchEnabled =
+    (capabilities && capabilities[chain.id]?.atomicBatch?.supported === true) ??
+    false;
+  const paymasterServiceEnabled =
+    (capabilities &&
+      capabilities[chain.id]?.paymasterService?.supported === true) ??
+    false;
+  const auxiliaryFundsEnabled =
+    (capabilities &&
+      capabilities[chain.id]?.auxiliaryFunds?.supported === true) ??
+    false;
+
+  return {
+    paymaster: paymasterServiceEnabled,
```

Comment:
why not just use the names from the ERCs and bubbling these up directly?

Reply:
edited it to just bubble up the capabilities names from the ERCs, 

cc @Zizzamia if we still want the `has` or `is` prefix pattern

I've also renamed the context key `capabilities` -> `walletCapabilities` more increased clarity

Reply:
Year using the the ERCs name was a good call.

I think to keep the same standard we have around OchainKit for boolean using the prefix `is` or `has`, I think it's important to stick with that.

Based on the context, seems like `hasSomething` seems the right prefix.